tests/conftest.py

```diff
@@ -5,3 +5,15 @@
 if 'tests' not in os.getcwd():
     os.chdir('./tests/')
     sys.path.append('../src/')
+
+from unittest.mock import MagicMock
+
+import pytest
+from discord.ext.commands import Context
+
+from main import Bot
+
+
+@pytest.fixture
+def ctx():
+    Context(prefix='durka', bot=MagicMock(spec=Bot), message=MagicMock())
```

Comment:
What does this test do?

Reply:
It's not a test, it's a fixure, a feature of pytest. After defining fixtures, we can use them directly in tests by creating arguments with the same name as the wanted fixture.

Reply:
A mock context is... well, a mock of a [context](https://discordpy.readthedocs.io/en/rewrite/ext/commands/api.html#context). It's only really used in tests to call the metric ton of methods which require a discord.py context passed as an argument.

The cool thing about fixtures is that a new instance is created each test, so that each instance can be tailored before the test for the test.

Reply:
I see - will read more into this. Thank you :)